packages/engine-server/src/drivers/file/storev2.ts

```diff
@@ -556,12 +556,25 @@ export class FileStorage implements DStore {
     const {
       notes,
       cacheUpdates,
+      fileMetaDict,
       errors: parseErrors,
     } = await new NoteParser({
       store: this,
       cache,
       logger: this.logger,
     }).parseFiles(noteFiles, vault);
+    const seenFiles = new Set(
+      _.values(fileMetaDict).flatMap((items) => {
+        return items.map((items) => {
+          return path.parse(items.fpath).name;
+        });
+      })
+    );
+    _.keys(cache.notes).map((key) => {
```

Comment:
instead of removing cache entries here, we already loop through all notes below. can we delete notes from this set as we process them - any notes not deleted should be considered stale and need to be deleted from the cache

Reply:
good point, fixed